src/common/resources/mod.rs

```diff
@@ -0,0 +1,20 @@
+pub mod allocation;
+pub mod descriptor;
+pub mod request;
+
+pub type NumOfCpus = u32;
+pub type CpuId = u32;
+
+/// Resource identification wrt. global resource list (stored in Core)
+pub type GenericResourceId = u32;
```

Comment:
It seems that mixing `GenericResourceId` and `GenericResourceIndex` could be very easy. It would be nice to add a comment to all these types or (ideally) use a newtype wrapper for them. I will add these wrappers in a follow-up PR.